addons/crm_livechat/models/mail_channel.py

```diff
@@ -26,16 +26,32 @@ def _execute_command_lead(self, **kwargs):
         self._send_transient_message(partner, msg)
 
     def _convert_visitor_to_lead(self, partner, channel_partners, key):
+        """ Create a lead from channel /lead command
+        :param partner: internal user partner (operator) that created the lead;
+        :param channel_partners: channel members;
+        :param key: operator input in chat ('/lead Lead about Product')
+        """
         description = ''.join(
             '%s: %s\n' % (message.author_id.name or self.anonymous_name, message.body)
             for message in self.channel_message_ids.sorted('id')
         )
+        # if public user is part of the chat: consider lead to be linked to an
+        # anonymous user whatever the participants. Otherwise keep only share
+        # partners (no user or portal user) to link to the lead.
+        customers = self.env['res.partner']
+        for customer in channel_partners.partner_id.filtered('active'):
+            if customer.partner_share and any(not user._is_public() for user in customer.user_ids) if customer.user_ids else True:
```

Comment:
This is a bit painful to read.
I'm pretty sure this does the same thing:
`if customer.partner_share and (not customer.user_ids or all(user._is_public() for user in customer.user_ids)):`

Reply:
Suggestion, do we really want to worry about groups or users here?
Wouldn't `self.env.ref('base.public_partner') not in customers` be enough?

Reply:
Nope because you may want to setup and use several public users in multi website environment.

Reply:
Can we ensure we always use `base.public_partner` in the first place?

Otherwise it would require our JS to know the list of all public partners of all website to work, which it currently doesn't do (but we can also add this information if necessary).

Reply:
Public partner ref direct use is technically not correct. You may unlink it and setup another public user / public partner. This is a common simplification done in code but may break as I think there is nothing preventing people from changing user_id on website model.

Reply:
Ok, we'll make sure to take the properly list of all potential public partners then.